Data/Base/Shaders/Materials/MaterialInterpolatorTemplate.h

```diff
@@ -0,0 +1,64 @@
+
+
+struct STAGE_TEMPLATE
+{
+  float4 Position : SV_Position;
+
+#if defined(USE_WORLDPOS)
+  float3 WorldPosition : WORLDPOS;
+#endif
+
+#if defined(USE_NORMAL)
+  float3 Normal : NORMAL;
+#endif
+
+#if defined(USE_TANGENT)
+  float3 Tangent : TANGENT;
+  float3 BiTangent : BITANGENT;
+#endif
+
+#if defined(USE_TEXCOORD0)
+  float2 TexCoord0 : TEXCOORD0;
+
+#  if defined(USE_TEXCOORD1)
+  float2 TexCoord1 : TEXCOORD1;
+#  endif
+#endif
+
+#if defined(USE_COLOR0)
+  float4 Color0 : COLOR0;
+
+#  if defined(USE_COLOR1)
+  float4 Color1 : COLOR1;
+#  endif
+#endif
+
+#if defined(USE_DEBUG_INTERPOLATOR)
+  float4 DebugInterpolator : DEBUG_INTERPOLATOR;
+#endif
+
+#if defined(CUSTOM_INTERPOLATOR)
+#  ifndef CUSTOM_INTERPOLATOR_NAME
```

Comment:
This does not really work in all cases. Sometimes you need more than one custom interpolator, so previously you could do something like 
#define CUSTOM_INTERPOLATOR float3 a : BLUBB; float3 b : BLABB;
I would suggest that we have a custom function that copies the data in the geometry shader:
CopyCustomInterpolators(GS_OUT output, VS_OUT input) and the user has to implement that if CUSTOM_INTERPOLATOR is defined. We could add a macro that implements the function for different numbers of interpolators used:
CUSTOM_INTERPOLATOR_COPY1(name1)
CUSTOM_INTERPOLATOR_COPY2(name1, name2)
etc.

Reply:
Good point! I've replaced the CUSTOM_INTERPOLATOR_NAME define with the CopyCustomInterpolators function. I didn't add CUSTOM_INTERPOLATOR_COPY1 etc as that would require an include before the #include <Shaders/Materials/MaterialStereoGeometryShader.h> which we don't have right now so it is easier to just define the function instead.

Reply:
Looks good for know. We can add a macro later if needed.